c10/core/TensorImpl.h

```diff
@@ -1318,6 +1318,26 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
     device_opt_ = storage_.device();
   }
 
+  virtual void update_strides_to_format(MemoryFormat memory_format) {
```

Comment:
This method looks new; if it is, where is the documentation?

Reply:
Also, some name bikeshedding: can we give this a name that more obviously implies something like "this trashes the data, don't use this on tensors with meaningful data on them"

Reply:
We also have `update_to_contiguous_strides`. Do you have good name suggestion for both?
How about `data_unsafe_update_strides_to_format` and merge `update_to_contiguous_strides` and `data_unsafe_update_strides_to_format` into one function?

Reply:
`empty_tensor_restride`? Combining those 2 functions into one seems like the right move.

Reply:
done